config/config_test.go

```diff
@@ -142,3 +144,62 @@ func TestMergeGlobalConfig(t *testing.T) {
 	// The log level should now be debug.
 	assert.Equal(t, "debug", config.Global.Loggers[Default].Level)
 }
+
+// initializeConfig initializes the configuration with the given context.
+// It returns a pointer to the Config struct. If configuration initialization fails,
+// the test will fail with an error message.
+func initializeConfig(ctx context.Context, t *testing.T) *Config {
+	t.Helper()
+	config := NewConfig(ctx, Config{
+		GlobalConfigFile: parentDir + GlobalConfigFilename,
+		PluginConfigFile: parentDir + PluginsConfigFilename,
+	})
+	err := config.InitConfig(ctx)
+	require.Nil(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```

Reply:
`config.InitConfig` returns a `errors.GatewayDError`, not a standard error. so, it’s an object, and we can assert it using require.NoError.

Reply:
So strange, I would have added the Error() string interface to this struct to solve this.

If you don't/cannot want to do this, you can simply rename the variable `err`, err is type connoted.

Use`res` or whatever

Reply:
GatewayDError is actually an implementation of an error, and we have Error() string. The issue is that our functions return `*gerr.GatewayDError` instead of the `error` type. When we return nil as *gerr.GatewayDError, it’s basically like returning (gerr.GatewayDError, nil), and require.NoError can’t handle that. This link explains it better: [Go FAQ - nil error](https://go.dev/doc/faq#nil_error).

The solution is to change all return types to the error type. I assume this is outside the scope of this PR.

Renaming it to something else isn’t a great idea because it’s really just an error. Later on, if we switch to using error, it’ll be easier to handle.